enhancements/builds/volume-secrets.md

```diff
@@ -0,0 +1,224 @@
+
+Future extensions to this enhancement could enable the use case of providing build input content from a persistent volume and allowing the build to store/cache content for future builds on such a volume.  Those will be discussed in the future enhancement at that time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+We will need to introduce a new mechanism in the build api which allows the user to indicate that they want to inject "volume" content into the build.  Initially the only allowed volume types will be configmaps and secrets.  The api will otherwise be similar to the existing secret/configmap injection api in which users identify the configmap/secret and the target path for injection.  
+
+This will be done by adding a Volume[] field to the Source and Docker strategy structs.  The Volume[] field will allow 
+defining volumes to be mounted to the build pod in the same way that a normal pod allows for this.  Similarly a VolumeMount[]
+field will be added, but without the MountPropagation and SubPath fields.  MountPropagation and SubPath can be considered
+for support in the future.
+
+These fields will be wired, via the build controller, directly to the build pod that is constructed, modulo some validation logic to constrain the types of Volumes we want to support (initially just configmaps and secrets).  In addition all mounts
+into the pod will be done at a path of our choosing, not the VolumeMount path specified, to ensure the user cannot 
+overwrite critical function inside the build pod and use it as an escalation pathway.
+
+The logic that invokes buildah will then pass the mounted directories as volume mount arguments.  The mount path provided
+to buildah will be determined from the VolumeMount specification.
+
+Proposed api/structs:
+Note:  DockerBuildStrategy will be updated in the same way.
+```
+// SourceBuildStrategy defines input parameters specific to an Source build.
+type SourceBuildStrategy struct {
+  // From is reference to an DockerImage, ImageStream, ImageStreamTag, or ImageStreamImage from which
+  // the docker image should be pulled
+  From kapi.ObjectReference
+
+  // PullSecret is the name of a Secret that would be used for setting up
+  // the authentication for pulling the Docker images from the private Docker
+  // registries
+  PullSecret *kapi.LocalObjectReference
+
+  // Env contains additional environment variables you want to pass into a builder container.
+  Env []kapi.EnvVar
+
+  // Scripts is the location of Source scripts
+  Scripts string
+
+  // Incremental flag forces the Source build to do incremental builds if true.
+  Incremental *bool
+
+  // ForcePull describes if the builder should pull the images from registry prior to building.
+  ForcePull bool
+
+  // Volumes is a list of volumes that can be mounted by the build
+  // More info: https://kubernetes.io/docs/concepts/storage/volumes
+  Volumes []kapi.Volume
```

Comment:
I think we would have validation rules that prohibit some volume types to start with?

Reply:
yes https://github.com/openshift/enhancements/pull/174/files#diff-5350e9cec1e0f35ccc67dcfbdec57bfaR70

Reply:
(godoc should be updated to reflect the supported set)

Reply:
(updated the godoc)

Reply:
See my comment above wrt providing our own object that is a subset of `kapi.Volume`.